multiversx_sdk/abi/serializer.py

```diff
@@ -0,0 +1,116 @@
+from typing import Any, List, Sequence
+
+from multiversx_sdk.abi.codec import Codec
+from multiversx_sdk.abi.parts import PartsHolder
+from multiversx_sdk.abi.values_multi import *
+
+
+class Serializer:
+    """
+    The serializer follows the rules of the MultiversX Serialization format:
+    https://docs.multiversx.com/developers/data/serialization-overview
+    """
+
+    def __init__(self, parts_separator: str):
+        if not parts_separator:
+            raise ValueError("cannot create serializer: parts separator must not be empty")
+
+        self.parts_separator = parts_separator
+        self.codec = Codec()
+
+    def serialize(self, input_values: Sequence[Any]) -> str:
+        parts = self.serialize_to_parts(input_values)
+        return self._encode_parts(parts)
+
+    def serialize_to_parts(self, input_values: Sequence[Any]) -> List[bytes]:
+        parts_holder = PartsHolder([])
+        self._do_serialize(parts_holder, input_values)
+        return parts_holder.get_parts()
+
+    def _do_serialize(self, parts_holder: PartsHolder, input_values: Sequence[Any]):
+        for i, value in enumerate(input_values):
+            if value is None:
+                raise ValueError("cannot serialize nil value")
```

Comment:
`nil` seems very Golang-ish

Reply:
Fixed in https://github.com/multiversx/mx-sdk-py/pull/49.